moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -83,18 +98,67 @@ class PlanningComponent
+                           T default_value)
+    {
+      // Try to get parameter or use default
+      if (!node->get_parameter_or(param_name, output_value, default_value))
+      {
+        RCLCPP_WARN(node->get_logger(),
+                    "Parameter \'%s\' not found in config use default value instead, check parameter type and "
+                    "namespace in YAML file",
+                    (param_name).c_str());
+      }
+    }
+
+    void load(const rclcpp::Node::SharedPtr& node, const std::string& param_namespace = "")
     {
+      // Set namespace
       std::string ns = "plan_request_params.";
-      node->get_parameter_or(ns + "planner_id", planner_id, std::string(""));
-      node->get_parameter_or(ns + "planning_pipeline", planning_pipeline, std::string(""));
-      node->get_parameter_or(ns + "planning_time", planning_time, 1.0);
-      node->get_parameter_or(ns + "planning_attempts", planning_attempts, 5);
-      node->get_parameter_or(ns + "max_velocity_scaling_factor", max_velocity_scaling_factor, 1.0);
-      node->get_parameter_or(ns + "max_acceleration_scaling_factor", max_acceleration_scaling_factor, 1.0);
+      if (!param_namespace.empty())
+      {
+        ns = param_namespace + ".plan_request_params.";
+      }
+
+      // Declare parameters
+      declareOrGetParam<std::string>(node, ns + "planner_id", planner_id, std::string(""));
+      declareOrGetParam<std::string>(node, ns + "planning_pipeline", planning_pipeline, std::string(""));
+      declareOrGetParam<double>(node, ns + "planning_time", planning_time, 1.0);
+      declareOrGetParam<int>(node, ns + "planning_attempts", planning_attempts, 5);
+      declareOrGetParam<double>(node, ns + "max_velocity_scaling_factor", max_velocity_scaling_factor, 1.0);
+      declareOrGetParam<double>(node, ns + "max_acceleration_scaling_factor", max_acceleration_scaling_factor, 1.0);
     }
   };
 
+  /// Planner parameters provided with the MotionPlanRequest
+  struct MultiPipelinePlanRequestParameters
+  {
+    MultiPipelinePlanRequestParameters(const rclcpp::Node::SharedPtr& node,
+                                       const std::vector<std::string>& planning_pipeline_names)
+    {
+      multi_plan_request_parameters.reserve(planning_pipeline_names.size());
+
+      for (auto const& planning_pipeline_name : planning_pipeline_names)
+      {
+        PlanRequestParameters parameters;
+        parameters.load(node, planning_pipeline_name);
+        multi_plan_request_parameters.push_back(parameters);
+      }
+    }
+    // Plan request parameters for the individual planning pipelines which run concurrently
+    std::vector<PlanRequestParameters> multi_plan_request_parameters;
+  };
+
+  /// \brief A solution callback function type for the parallel planning API of planning component
+  typedef std::function<planning_interface::MotionPlanResponse(
+      std::vector<planning_interface::MotionPlanResponse> const& solutions)>
```

Comment:
I don't think that we should mix east const into here, better be consistent with the rest. I'm open for discussions to change the style overall, though.